python/pyarrow/tests/test_parquet.py

```diff
@@ -2008,6 +2008,19 @@ def test_filters_inclusive_set(tempdir, use_legacy_dataset):
     assert 'c' not in result_df['string'].values
     assert False not in result_df['boolean'].values
 
+    dataset = pq.ParquetDataset(
+        base_path, filesystem=fs,
+        filters=[('integer', 'in', [1]), ('string', 'in', ('a', 'b')),
+                 ('boolean', 'not in', {False})],
+        use_legacy_dataset=use_legacy_dataset
+    )
+    table = dataset.read()
+    result_df = (table.to_pandas().reset_index(drop=True))
+
+    assert 0 not in result_df['integer'].values
+    assert 'c' not in result_df['string'].values
+    assert False not in result_df['boolean'].values
+
```

Comment:
Can you also add a test to assert the error you get when passing an invalid "in" filter (like `("col", "in", 4)`)

Reply:
There is a test case: [('integers', 'in', 3), ]. It will raise a TypeError as in the new dataset API.